### DISCUSSION TITLE
Import and Export

### DISCUSSION THREAD

**Discussion by rwl7532** [jkotlinski/durexforth#536] [Q&A] [answered]
I see lots of code examples here in various discussions.  How is the import or export handled? Whether my code is on the SD card that my Commodore uses or reading the same card on my laptop using VICE I don't know how to get my code in or out. Taking a picture of the screen is all I have currently.

**Comment by jkotlinski** [comment]
For short pieces of code, I usually transfer using copy/paste. Ctrl+C in
browser, then paste into Vice.

tors 30 mars 2023 kl. 18:35 skrev rwl7532 ***@***.***>:

**Reply by rwl7532** [reply]
For VICE on my Windows laptop, I can begin editing with I but what key press is the the equivalent of the key to the left of !/1 that gets me out of insert mode? Esc not working.

**Reply by Whammo** [reply, marked as answer]
end

**Reply by rwl7532** [reply]
I'll try it.~ or ` was the answer from the internet.
Thanks. I'm close to being free!

**Comment by ekipan** [comment]
Once you have a program in `v` you might want to `:w`rite it to a .d64 disk image file. VICE comes with the c1541 tool to edit disks directly, here's a [Youtube video covering some basics](https://www.youtube.com/watch?v=javD5O5hlEc). There's other options, like [DirMaster](https://style64.org/dirmaster) for Windows with a GUI interface.

**Reply by ekipan** [reply]
Whenever I want to update my block editor gist I've settled on this workflow:

1. Open the disk image in DirMaster.
2. Select the `z` file.
3. Menu: `File > View > SEQ` then `File > Export`
4. Filetype `ASCII` Filename `z.fs`
5. Delete a couple junk characters at the start and end of file then copypaste into Github Gists.

You can select all these menu options with the Alt key so this ends up taking like 10 seconds, easier than my previous workflow of paging through and copypasting individual screens in `v`.
